core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -598,22 +598,28 @@ class AdminManager(val config: KafkaConfig,
     }
 
     alterConfigOps.foreach { alterConfigOp =>
+      val configPropName = alterConfigOp.configEntry().name()
       alterConfigOp.opType() match {
         case OpType.SET => configProps.setProperty(alterConfigOp.configEntry().name(), alterConfigOp.configEntry().value())
         case OpType.DELETE => configProps.remove(alterConfigOp.configEntry().name())
         case OpType.APPEND => {
           if (!listType(alterConfigOp.configEntry().name(), configKeys))
             throw new InvalidRequestException(s"Config value append is not allowed for config key: ${alterConfigOp.configEntry().name()}")
-          val oldValueList = configProps.getProperty(alterConfigOp.configEntry().name()).split(",").toList
+          val oldValueList = configProps.getProperty(configPropName,
+            ConfigDef.convertToString(configKeys(configPropName).defaultValue, ConfigDef.Type.LIST)).split(",").toList
```

Comment:
`configKeys(configPropName).defaultValue` can return `null`. we may need to treat  `null' as empty string,

Reply:
Good point.